ci-operator/step-registry/cucushift/hypershift-extended/capi/health-check/cucushift-hypershift-extended-capi-health-check-commands.sh

```diff
@@ -0,0 +1,246 @@
+#!/bin/bash
+
+set -xeuo pipefail
+
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "setting the proxy"
+        # cat "${SHARED_DIR}/proxy-conf.sh"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+function rosa_login() {
+    # default stage api url
+    ocm_api_url="https://api.stage.openshift.com"
+    if [[ "${OCM_LOGIN_ENV}" == "production" ]] ; then
+     ocm_api_url="https://api.openshift.com"
+    elif [[ "${OCM_LOGIN_ENV}" == "integration" ]] ; then
+     ocm_api_url="https://api.integration.openshift.com"
+    fi
+
+    ROSA_VERSION=$(rosa version)
+    ROSA_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
+
+    if [[ ! -z "${ROSA_TOKEN}" ]]; then
+      echo "Logging into ${OCM_LOGIN_ENV} with offline token using rosa cli ${ROSA_VERSION}"
+      rosa login --env "${OCM_LOGIN_ENV}" --token "${ROSA_TOKEN}"
+      ocm login --url "${OCM_LOGIN_ENV}" --token "${ROSA_TOKEN}"
+    else
+      echo "Cannot login! You need to specify the offline token ROSA_TOKEN!"
+      exit 1
+    fi
+}
+
+function check_rosacontrolplane() {
+    local name=$1
+    local namesapce=$2
+
+}
+
+set_proxy
+rosa_login
+
+export KUBECONFIG="${SHARED_DIR}/kubeconfig"
+if [[ -f "${SHARED_DIR}/mgmt_kubeconfig" ]]; then
+  export KUBECONFIG="${SHARED_DIR}/mgmt_kubeconfig"
+fi
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+export AWS_REGION=${REGION}
+export AWS_PAGER=""
+
+# get cluster namesapce
+CLUSTER_NAME=$(cat "${SHARED_DIR}/cluster-name")
+if [[ -z "${CLUSTER_NAME}" ]] ; then
+  echo "Error: cluster name not found"
+  exit 1
+fi
+
+read -r namespace _ _  <<< "$(oc get cluster -A | grep ${CLUSTER_NAME})"
+if [[ -z "${namespace}" ]]; then
+  echo "capi cluster name not found error, ${CLUSTER_NAME}"
+  exit 1
+fi
+
+echo "check cluster, rosacluster status"
+cluster_status=$(oc get cluster "${CLUSTER_NAME}" -n "${namespace}" -ojsonpath='{.status.phase}')
```

Comment:
Maybe storing `oc get cluster "${CLUSTER_NAME}" -n "${namespace}" -o json` in a temporary file could speed up and reduce potential failing risk.

Reply:
No need for the cluster resource, because this is the only one to check its status.